File: lib/base/base_bot.py
class BaseBot:
    def handle_new_round(self, game_state, round_state, active):
        raise NotImplementedError('handle_new_round')

    def handle_round_over(self, game_state, terminal_state, active):
        raise NotImplementedError('handle_round_over')

    def get_action(self, game_state, round_state, active):
        raise NotImplementedError('get_action')

File: rust/src/lib/base/base_bot.rs
use crate::lib::game::poker_moves::PokerMove;
use crate::lib::game::poker_state::{GameState, RoundState, TerminalState};

pub trait BaseBot {
    fn handle_new_round(&mut self, game_state: &GameState, round_state: &RoundState, active: usize);
    
    fn handle_round_over(&mut self, game_state: &GameState, terminal_state: &TerminalState, active: usize);
    
    fn get_action(&mut self, game_state: &GameState, round_state: &RoundState, active: usize) -> PokerMove;
}